utils/sampler/rand.go

```diff
@@ -4,58 +4,92 @@
 package sampler
 
 import (
-	"math/rand"
+	"math"
 	"sync"
 	"time"
 
 	"gonum.org/v1/gonum/mathext/prng"
 )
 
-var (
-	int63Mask uint64 = 1<<63 - 1
-	globalRNG        = newRNG()
-)
+var globalRNG = newRNG()
 
-func newRNG() rng {
+func newRNG() *rng {
 	source := prng.NewMT19937()
 	source.Seed(uint64(time.Now().UnixNano()))
-	// We don't use a cryptographically secure source of randomness here, as
-	// there's no need to ensure a truly random sampling.
-	return rand.New(&syncSource{rng: source}) // #nosec G404
+	return &rng{rng: source}
```

Comment:
Removing the comment suggests you will use this PRNG as a cryptographically secure one. I recommend leaving the comment here. It will also raise questions in future audits.

Reply:
ok